x-pack/legacy/plugins/epm/public/screens/detail/header.tsx

```diff
@@ -3,52 +3,76 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-
-import React, { Fragment, useCallback, useState } from 'react';
-import {
-  EuiButton,
-  EuiButtonEmpty,
-  EuiButtonEmptyProps,
-  EuiFlexGroup,
-  EuiFlexItem,
-  EuiPage,
-  EuiTitle,
-  IconType,
-} from '@elastic/eui';
+import React, { Fragment, useCallback, useState, useMemo } from 'react';
 import styled from 'styled-components';
-// TODO: either
-// * import from a shared point
-// * duplicate inside our folder
-import { useTrackedPromise } from '../../../../infra/public/utils/use_tracked_promise';
-import { installPackage } from '../../data';
+import { EuiFlexGroup, EuiFlexItem, EuiPage, EuiTitle, IconType } from '@elastic/eui';
 import { PLUGIN } from '../../../common/constants';
 import { PackageInfo } from '../../../common/types';
 import { Version } from '../../components/version';
 import { IconPanel } from '../../components/icon_panel';
-import { useBreadcrumbs, useCore, useLinks } from '../../hooks';
+import { useBreadcrumbs, useLinks } from '../../hooks';
 import { CenterColumn, LeftColumn, RightColumn } from './layout';
+import { InstallationButton } from '../../components/installation_button';
+import { ConfirmPackageInstall } from '../../components/confirm_package_install';
+// TODO: either
+// * import from a shared point
+// * duplicate inside our folder
+import { useTrackedPromise } from '../../../../infra/public/utils/use_tracked_promise';
+import { installPackage } from '../../data';
+import { NavButtonBack } from '../../components/nav_button_back';
```

Comment:
Can you move the 3 components extracted to `../../components/` back into this file? I'd prefer to wait until we actually use them in more than 1 place before moving them there. Yes, by that logic, some of the items already in that directory shouldn't be there.

Reply:
I dont think the reason for putting something in its own component is solely because its going to be used more than once. Here it's a separation of concerns and it's easier to read.  This file was getting too large and I think it's easier for a developer to read when there aren't so many components in one file.

Reply:
True. I combined two thoughts into one imprecise one.

Can you keep those separate files in the `screens/details` folder instead of lifting them up too `../../components`?

We do this elsewhere for the reasons you mentioned.